src/V3Param.cpp

```diff
@@ -101,14 +104,32 @@ class ParameterizedHierBlocks final {
                 const bool inserted = consts.emplace(pIt->first, std::move(constp)).second;
                 UASSERT(inserted, pIt->first << " is already added");
             }
+            // origName may be already registered, but it's fine.
+            m_modParams.insert({hierOpt.second.origName(), {}});
         }
         for (AstNodeModule* modp = nodep->modulesp(); modp;
              modp = VN_CAST(modp->nextp(), NodeModule)) {
             if (hierOpts.find(modp->prettyName()) != hierOpts.end()) {
                 m_hierBlockMod.emplace(modp->name(), modp);
             }
+            auto defParam = m_modParams.find(modp->name());
+            if (defParam != m_modParams.end()) {
+                // modp is the original of parameterized hierarchical block
+                for (AstNode* stmtp = modp->stmtsp(); stmtp; stmtp = stmtp->nextp()) {
+                    if (AstVar* varp = VN_CAST(stmtp, Var)) {
+                        if (varp->isGParam()) defParam->second.emplace(varp->name(), varp);
+                    }
+                }
+            }
         }
     }
+    template <class MAP, typename KEY>
```

Comment:
```suggestion
    template <class T_MAP, typename T_KEY>
```

However I'm not sure this is worth it. You use it only twice, it saves really only one line of code, and discourages making a nicer assert message - and the error message will point to this function rather than the place that is calling it - which seems less useful.

Reply:
Indeed.
Removed in 498bb35042bca94eb41bab569e1b1534b861b3cb.